java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowWriter.java

```diff
@@ -174,6 +178,12 @@ public long bytesWritten() {
     return out.getCurrentPosition();
   }
 
+  private CompressionCodec getCodec() {
```

Comment:
It seems we should be able to initialize the codec once and reuse it in the constructor, rather than add all the new fields?

Reply:
@lidavidm an issue was filed here: https://github.com/apache/arrow/issues/39222
Also, I can work on this.

Reply:
Again: why did we pull this out? It's called only once. Why are we adding a bunch of new fields? We don't use them.